ai_ta_backend/nomic_logging.py

```diff
@@ -32,9 +32,13 @@ def giveup_hdlr(e):
   conversation = e_args['conversation']
   print("giveup_hdlr() called with exception:", e_str)
   if e_str == 'You must specify a unique_id_field when creating a new project.':

```

Comment:
@KastanDay I have added exception handling in this `giveup_hdlr()` to trigger the `create_nomic_map()` during specific exceptions. Yet, the overall function raises an ERROR at the end and logs it to Sentry. Any suggestions on how to go around this?
<img width="942" alt="image" src="https://github.com/UIUC-Chatbot/ai-ta-backend/assets/25609228/a7045f5c-1e3f-4c2e-9e17-51d3cb132461">

Reply:
@star-nox I think the root cause is you have to catch the exception. 

Something like this. This way you capture all exceptions, then handle the if-else _inside_ the exception block.
```python 
try:
    result = create_nomic_map(course_name, conversation)
except Exception as e:
    if str(e) == 'You must specify a unique_id_field when creating a new project.':
        # Handle this specific exception
        pass
    else:
        # Handle all other exceptions
        pass
```

Reply:
@KastanDay I added exception handling in the main logging function itself and it worked! Its ready to merge.